### ISSUE TITLE
UnhandledPromiseRejectionWarning for locally running app

### ISSUE THREAD

**Issue by jcteague** [DevExpress/testcafe#922]
### Are you requesting a feature or reporting a bug?
bug

### What is the current behavior?
Page is not loading and getting an error message on console
```(node:98695) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: First argument must be a string or Buffer```

### What is the expected behavior?
to load the page and run the test

### How would you reproduce the current behavior (if this is a bug)?
I'm running a react website locally. I'm using SimpleHttpServer to run from the project root.
I can see from server output it is requesting both the page and the javacript.

The page never loads from the browser opened by testcafe and I'm getting the error message on the console

from the same project I've got a smoke test that opens `http://devexpress.github.io/testcafe/example`
and that will open and run. 

#### Provide the test code and the tested page URL (if applicable)

Tested page URL:

Test code

```js
import { Selector } from 'testcafe';

import { expect } from 'chai';
fixture `Welcome Screen`
  .page('http://localhost:8080/#');
const page = {
  welcomeBanner: Selector('.textGraph h1')
};

test('should show welcome banner', async t => {
  await t.wait(1000);
  console.log('starting test');
  let banner = await Selector('.textGraph h1');
  console.log(banner.innerText);
  expect(banner.innerText).to.equal('Welcome');

});
```
The console output is this:

```
testcafe chrome test/uat/welcome-screen/welcome-screen-tests.js

 Running tests in:
  Chrome 53.0.2785 / Mac OS X 10.12.0

 Welcome Screen
(node:99056) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: First argument must be a string or Buffer
starting test
undefined
```

### Specify your

* operating system: mac
* testcafe version: 0.9.0
* node.js version: v6.8.0

**Comment by inikulin** [comment]
Hi John.

Thank you for the report. Can you run
```sh
DEBUG=testcafe:test-run:* testcafe {your-browser} {your-test-files}
```
and post the output here, please?

**Comment by jcteague** [comment]
```
DEBUG=testcafe:test-run:* testcafe chrome test/uat/welcome-screen/welcome-screen-tests.js

 Running tests in:
 - Chrome 53.0.2785 / Mac OS X 10.12.0

 Welcome Screen
  testcafe:test-run:Chrome 53.0.2785 / Mac OS X 10.12.0:command 
    {
      "type": "wait",
      "timeout": 1000
    }
 +0ms
(node:99763) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: First argument must be a string or Buffer
  testcafe:test-run:Chrome 53.0.2785 / Mac OS X 10.12.0:driver-message 
    {
      "cmd": "ready",
      "status": {
        "id": "286.18000000000006",
        "isCommandResult": false,
        "executionError": null,
        "pageError": null,
        "resent": false,
        "result": null
      },
      "disableResending": true,
      "sessionId": "HketkOYCJg"
    }
 +835ms
starting test
undefined
  testcafe:test-run:Chrome 53.0.2785 / Mac OS X 10.12.0:command 
    {
      "type": "test-done"
    }
 +182ms
```

**Comment by jcteague** [comment]
This must have something to do with the react app itself.  I tested against a deployed instance.  On the deployed instance, it must go through a login screen.  I changed the test to input the login screen fields and then navigate to the url that runs the react app.  The login works fine, but I get the same error after I navigate to my application.

We are using hashHistory for the react app, so it appends `#<random-key>` to the url.  Not sure if that matters.

here is the debug output
```
 Welcome Screen
  testcafe:test-run:Chrome 53.0.2785 / Mac OS X 10.12.0:command 
    {
      "type": "type-text",
      "selector": {
        "type": "execute-selector",
        "instantiationCallsiteName": "Selector",
        "fnCode": "(function(){\n   var __f$=(function(){return document.querySelectorAll(\"#username\");});\n   return function(){\n       return window['%testCafeSelectorFilter%'](__f$.apply(this, arguments), __dependencies$.__filterOptions$);\n   };\n})();",
        "args": [
          []
        ],
        "dependencies": [
          {
            "__filterOptions$": {
              "index": 0,
              "text": {
                "@t": "[[undefined]]",
                "data": ""
              }
            }
          }
        ],
        "visibilityCheck": true,
        "timeout": null,
        "index": 0
      },
      "text": "jon",
      "options": {
        "offsetX": null,
        "offsetY": null,
        "modifiers": {
          "ctrl": false,
          "alt": false,
          "shift": false,
          "meta": false
        },
        "caretPos": null,
        "replace": false
      }
    }
 +0ms
  testcafe:test-run:Chrome 53.0.2785 / Mac OS X 10.12.0:driver-message 
    {
      "cmd": "ready",
      "status": {
        "id": "1036.6850000000002",
        "isCommandResult": false,
        "executionError": null,
        "pageError": null,
        "resent": false,
        "result": null
      },
      "disableResending": true,
      "sessionId": "HJ3_hM1el"
    }
 +1s
  testcafe:test-run:Chrome 53.0.2785 / Mac OS X 10.12.0:driver-message 
    {
      "cmd": "ready",
      "status": {
        "id": "3131.55",
        "isCommandResult": true,
        "executionError": null,
        "pageError": null,
        "resent": false,
        "result": null
      },
      "disableResending": true,
      "sessionId": "HJ3_hM1el"
    }
 +2s
  testcafe:test-run:Chrome 53.0.2785 / Mac OS X 10.12.0:command 
    {
      "type": "type-text",
      "selector": {
        "type": "execute-selector",
        "instantiationCallsiteName": "Selector",
        "fnCode": "(function(){\n   var __f$=(function(){return document.querySelectorAll(\"#password\");});\n   return function(){\n       return window['%testCafeSelectorFilter%'](__f$.apply(this, arguments), __dependencies$.__filterOptions$);\n   };\n})();",
        "args": [
          []
        ],
        "dependencies": [
          {
            "__filterOptions$": {
              "index": 0,
              "text": {
                "@t": "[[undefined]]",
                "data": ""
              }
            }
          }
        ],
        "visibilityCheck": true,
        "timeout": null,
        "index": 0
      },
      "text": "fasterthansixmill",
      "options": {
        "offsetX": null,
        "offsetY": null,
        "modifiers": {
          "ctrl": false,
          "alt": false,
          "shift": false,
          "meta": false
        },
        "caretPos": null,
        "replace": false
      }
    }
 +1ms
  testcafe:test-run:Chrome 53.0.2785 / Mac OS X 10.12.0:driver-message 
    {
      "cmd": "ready",
      "status": {
        "id": "3902.135",
        "isCommandResult": true,
        "executionError": null,
        "pageError": null,
        "resent": false,
        "result": null
      },
      "disableResending": true,
      "sessionId": "HJ3_hM1el"
    }
 +770ms
  testcafe:test-run:Chrome 53.0.2785 / Mac OS X 10.12.0:command 
    {
      "type": "click",
      "selector": {
        "type": "execute-selector",
        "instantiationCallsiteName": "Selector",
        "fnCode": "(function(){\n   var __f$=(function(){return document.querySelectorAll(\"#submitButton\");});\n   return function(){\n       return window['%testCafeSelectorFilter%'](__f$.apply(this, arguments), __dependencies$.__filterOptions$);\n   };\n})();",
        "args": [
          []
        ],
        "dependencies": [
          {
            "__filterOptions$": {
              "index": 0,
              "text": {
                "@t": "[[undefined]]",
                "data": ""
              }
            }
          }
        ],
        "visibilityCheck": true,
        "timeout": null,
        "index": 0
      },
      "options": {
        "offsetX": null,
        "offsetY": null,
        "modifiers": {
          "ctrl": false,
          "alt": false,
          "shift": false,
          "meta": false
        },
        "caretPos": null
      }
    }
 +1ms
  testcafe:test-run:Chrome 53.0.2785 / Mac OS X 10.12.0:driver-message 
    {
      "cmd": "wait-for-file-download",
      "sessionId": "HJ3_hM1el"
    }
 +1s
  testcafe:test-run:Chrome 53.0.2785 / Mac OS X 10.12.0:driver-message 
    {
      "cmd": "ready",
      "status": {
        "id": "1338.98",
        "isCommandResult": true,
        "executionError": null,
        "pageError": null,
        "resent": false,
        "result": null
      },
      "disableResending": true,
      "sessionId": "HJ3_hM1el"
    }
 +379ms
  testcafe:test-run:Chrome 53.0.2785 / Mac OS X 10.12.0:command 
    {
      "type": "navigate-to",
      "url": "http://pfhack3.saplinglearning.me/ibiscms/course/view.php?id=56105#/?_k=sxx64x"
    }
 +0ms
  testcafe:test-run:Chrome 53.0.2785 / Mac OS X 10.12.0:driver-message 
    {
      "cmd": "ready",
      "status": {
        "id": "376.93000000000006",
        "isCommandResult": true,
        "executionError": null,
        "pageError": null,
        "resent": false,
        "result": null
      },
      "disableResending": true,
      "sessionId": "HJ3_hM1el"
    }
 +2s
starting test
undefined
  testcafe:test-run:Chrome 53.0.2785 / Mac OS X 10.12.0:command 
    {
      "type": "test-done"
    }
 +20ms
(node:2729) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: First argument must be a string or Buffer
```

**Comment by AlexanderMoskovkin** [comment]
Hi @jcteague , thanks for your information.

It's hard to determine the cause of the problem without a page. Could you provide us a public link to your page where we can reproduce the problem? To save the confidential information (login, password), you can send it on [my email](mailto:alexander.moskovkin@devexpress.com). If necessary we can sign NDA, so we'll be able to debug your page.

Alternatively you can provide us a simplified page that reproduces the problem, so it doesn't contain any sensitive data.

If it's not possible, we can try to determine the cause of the issue remotely, but it will require some time and assistance from your side

Please, let me know what works best for you.

**Comment by jcteague** [comment]
I'm going to close this issue. But I still think there is something here, but I can't reproduce in test project and I solved the problem in actual project.

I solved the problem by updating to react-router@3.0.0.  hashHistory does not put the additional query string key on the url, and without it testcafe will run the test.

However, I tried using the older version of react-router that does use the additional parameter, but could not reproduce in the test project.

So while by gut feeling is it has something to do with that url param, I cannot reproduce enough to confirm.

Thanks
